_docs/guides/endpoints.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Install Istio for Google Cloud Endpoints Services  
+overview: Guideline to manually integrate Google Cloud Endpoints services with Istio.
+
+order: 42
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+This document provides a guideline to manually integrate Istio with existing
+Google Cloud Endpoints services. 
+
+## Before you begin
+
+You need to follow the instruction to setup an Endpoints service on GKE [instructions](https://cloud.google.com/endpoints/docs/openapi/get-started-kubernetes-engine)
+After setup, you should be able to get an API key and store it in `ENDPOINTS_KEY` environment variable and the external IP address `EXTERNAL_IP`.
+You may test the service using the following command: 
+```
+curl --request POST       --header "content-type:application/json"       --data '{"message":"hello world"}'       "http://${EXTERNAL_IP}:80/echo?key=${ENDPOINTS_KEY}"
+```
+
+You need to install Istio with [instructions]({{home}}/docs/setup/kubernetes/quick-start.html#google-kubernetes-engine).
+
+## HTTP Endpoints service.
```

Comment:
the headers typically don't terminate with periods.  See line 14 of your PR as an example.